### PR TITLE
Agora snapshot comments [GAWB-2032]

### PR DESCRIPTION
**Pull request by aednichols**
Requires https://github.com/broadinstitute/agora/pull/214

Have you read [CONTRIBUTING.md](../CONTRIBUTING.md) lately? If not, do that first.

I, the developer opening this PR, do solemnly pinky swear that:

- [ ] I've followed [the instructions](https://github.com/broadinstitute/firecloud-orchestration/blob/develop/CONTRIBUTING.md#api-changes) if I've made any changes to the API, _especially_ if they're breaking changes
- [ ] I've updated the RC_XXX release ticket with any manual steps required to release this change
- [ ] I've updated the [FISMA documentation](https://github.com/broadinstitute/firecloud-orchestration/blob/develop/CONTRIBUTING.md#fisma-documentation-changes) if I've made any security-related changes, including auth, encryption, or auditing

In all cases:

- [ ] Get two thumbsworth of review and PO signoff if necessary
- [ ] Verify all tests go green
- [ ] Squash and merge; you can delete your branch after this **unless it's for a hotfix**. In that case, don't delete it!
- [ ] Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

File: src/main/resources/swagger/api-docs.yaml [hunk 1]
```diff
@@ -221,6 +221,10 @@ paths:
           description: Snapshot ID of configuration.
           required: false
           type: number
+        - name: snapshotComment
```
Comment (rushtong) [inline src/main/resources/swagger/api-docs.yaml:224]: This doesn't appear to be `required`, similar to the Agora PR. Can this be annotated similarly?

**Review by rushtong — APPROVED** [review]
👍  One minor suggestion inline.

File: src/main/resources/swagger/api-docs.yaml (hunk 1 above)
Comment (rushtong) [reply src/main/resources/swagger/api-docs.yaml:224]: My bad, this is required - ignore comment. Pretty sure this also constitutes a breaking change to the API - we will need to handle that with a release email.

Comment (aednichols) [reply src/main/resources/swagger/api-docs.yaml:224]: It should not be `required`, snapshot comments are designed to be very optional. Will fix & check consistency across PRs.

Comment (rushtong) [reply src/main/resources/swagger/api-docs.yaml:224]: 👍  - Yes, as I dig more into the agora PR, it is becoming more clear, thanks!

File: src/main/resources/swagger/api-docs.yaml [hunk 2]
```diff
@@ -975,6 +979,10 @@ paths:
           description: Snapshot ID of method.
           required: false
           type: number
+        - name: snapshotComment
+          in: query
+          type: string
+          description: Snapshot comment of AgoraEntity
         - name: synopsis
           in: query
           description: Synopsis of method.
```
Comment (davidangb) [inline src/main/resources/swagger/api-docs.yaml]: need to add snapshotComment to the `POST /api/methods/{namespace}/{name}/{snapshotId}` endpoint, around line 1192. That's the copy-snapshot endpoint.

**Review by davidangb — COMMENTED** [review]
did not test; code looks good modulo one comment inline.

File: src/main/resources/swagger/api-docs.yaml (hunk 1 above)
Comment (aednichols) [reply src/main/resources/swagger/api-docs.yaml:224]: The default value of `required` is `false`, so this was correct, but I'm going to make it explicit.

**Review by davidangb — APPROVED** [review]
👍 updates look good to me. I am not at a place where I can run this locally yet, sorry!

**Review by rushtong — APPROVED** [review]
👍  LGTM

**Pull request merged**